src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -62,6 +70,53 @@ class TasksController(
     }
   }
 
+  private def deleteTasks()(implicit identity: Identity): Route = {
+    def doKill(force: Boolean, scale: Boolean, wipe: Boolean, tasksIdToAppId: Map[Id, PathId]) = {
+      def isAuthorized(tasksIdToAppId: Map[Id, PathId]): Boolean = tasksIdToAppId.values.exists(id => !authorizer.isAuthorized(identity, UpdateRunSpec, id))
+
+      // TODO authorization check is performed twice - in the taskKiller and also here, this one must remain
+      if (isAuthorized(tasksIdToAppId)) {
```

Comment:
this is a change I introduced in comparison to the old implementation. For all kill methods, authorization is checked in TaskKiller. For kill (not kill and scale) some time ago, duplicit authorization was introduced because we need to check authorization for all tasks before we start killing anything.

The change I introduced (which I hope should be fine) is that the authorization is checked in advance also for kill and scale (not just for kill).

Reply:
Is this still a todo then? What needs to be done?

Reply:
I talked with the author of the original todo and it was meant to point out a potential performance issue with the fact that the authorization is checked twice (once in controller and then somewhere deep in task killer). So I kept the comment in place if we ever want to refactor it (but it was totally out of scope of this PR)

The change that I made on top what was there before is that now the check is done twice for kill as well as killAndScale (previously it was for some reason just kill)